diff --git a/node_modules/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt b/node_modules/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt
index 1dca391..e4c8565 100644
--- a/node_modules/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt
+++ b/node_modules/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt
@@ -58,6 +58,10 @@ class CellularModule : Module() {
       telephonyManager()?.simOperator?.substring(3)
     }
 
+    AsyncFunction("getSignalStrengthAsync") {
+      telephonyManager()?.signalStrength?.level
+    }
+
     AsyncFunction("requestPermissionsAsync") { promise: Promise ->
       Permissions.askForPermissionsWithPermissionsManager(
         permissionsManager,
diff --git a/node_modules/expo-cellular/build/Cellular.d.ts b/node_modules/expo-cellular/build/Cellular.d.ts
index 033e593..69d224f 100644
--- a/node_modules/expo-cellular/build/Cellular.d.ts
+++ b/node_modules/expo-cellular/build/Cellular.d.ts
@@ -200,6 +200,19 @@ export declare function getMobileCountryCodeAsync(): Promise<string | null>;
  * ```
  */
 export declare function getMobileNetworkCodeAsync(): Promise<string | null>;
+/**
+ * @return Returns the signal strength value of the user's current network.
+ *
+ * On Android, the value will be a number between 0 and 4.
+ *
+ * On iOS and the web, this returns `null`.
+ *
+ * @example
+ * ```ts
+ * await Cellular.getSignalStrengthAsync(); // 3
+ * ```
+ */
+export async function getSignalStrengthAsync(): Promise<number | null>;
 /**
  * Checks user's permissions for accessing phone state.
  */
diff --git a/node_modules/expo-cellular/build/Cellular.js b/node_modules/expo-cellular/build/Cellular.js
index 2b4a68d..7c550fd 100644
--- a/node_modules/expo-cellular/build/Cellular.js
+++ b/node_modules/expo-cellular/build/Cellular.js
@@ -242,6 +242,24 @@ export async function getMobileNetworkCodeAsync() {
     }
     return await ExpoCellular.getMobileNetworkCodeAsync();
 }
+/**
+ * @return Returns the signal strength value of the user's current network.
+ *
+ * On Android, the value will be a number between 0 and 4.
+ *
+ * On iOS and the web, this returns `null`.
+ *
+ * @example
+ * ```ts
+ * await Cellular.getSignalStrengthAsync(); // 3
+ * ```
+ */
+export async function getSignalStrengthAsync() {
+  if (!ExpoCellular.getSignalStrengthAsync) {
+    throw new UnavailabilityError('expo-cellular', 'getSignalStrengthAsync');
+  }
+  return await ExpoCellular.getSignalStrengthAsync();
+}
 /**
  * Checks user's permissions for accessing phone state.
  */
diff --git a/node_modules/expo-cellular/build/ExpoCellular.web.d.ts b/node_modules/expo-cellular/build/ExpoCellular.web.d.ts
index 45484aa..2cdda17 100644
--- a/node_modules/expo-cellular/build/ExpoCellular.web.d.ts
+++ b/node_modules/expo-cellular/build/ExpoCellular.web.d.ts
@@ -11,6 +11,7 @@ declare const _default: {
     getCarrierNameAsync(): Promise<string | null>;
     getMobileCountryCodeAsync(): Promise<string | null>;
     getMobileNetworkCodeAsync(): Promise<string | null>;
+    getSignalStrengthAsync(): Promise<number | null>;
 };
 export default _default;
 //# sourceMappingURL=ExpoCellular.web.d.ts.map
\ No newline at end of file
diff --git a/node_modules/expo-cellular/build/ExpoCellular.web.js b/node_modules/expo-cellular/build/ExpoCellular.web.js
index 32d0234..6fc48a2 100644
--- a/node_modules/expo-cellular/build/ExpoCellular.web.js
+++ b/node_modules/expo-cellular/build/ExpoCellular.web.js
@@ -49,5 +49,8 @@ export default {
     async getMobileNetworkCodeAsync() {
         return null;
     },
+	async getSignalStrengthAsync() {
+        return null;
+	},
 };
 //# sourceMappingURL=ExpoCellular.web.js.map
\ No newline at end of file
diff --git a/node_modules/expo-cellular/ios/CellularModule.swift b/node_modules/expo-cellular/ios/CellularModule.swift
index a07cd62..f739d6b 100644
--- a/node_modules/expo-cellular/ios/CellularModule.swift
+++ b/node_modules/expo-cellular/ios/CellularModule.swift
@@ -32,6 +32,10 @@ public class CellularModule: Module {
     AsyncFunction("getMobileNetworkCodeAsync") { () -> String? in
       Self.currentCarrier()?.mobileNetworkCode
     }
+
+    AsyncFunction("getSignalStrengthAsync") { () -> Int? in
+      nil
+    }
   }
 
   // MARK: - internals
diff --git a/node_modules/expo-cellular/src/Cellular.ts b/node_modules/expo-cellular/src/Cellular.ts
index 2a12a2d..9d5a11c 100644
--- a/node_modules/expo-cellular/src/Cellular.ts
+++ b/node_modules/expo-cellular/src/Cellular.ts
@@ -261,6 +261,25 @@ export async function getMobileNetworkCodeAsync(): Promise<string | null> {
   return await ExpoCellular.getMobileNetworkCodeAsync();
 }
 
+/**
+ * @return Returns the signal strength value of the user's current network.
+ *
+ * On Android, the value will be a number between 0 and 4.
+ *
+ * On iOS and the web, this returns `null`.
+ *
+ * @example
+ * ```ts
+ * await Cellular.getSignalStrengthAsync(); // 3
+ * ```
+ */
+export async function getSignalStrengthAsync() {
+  if (!ExpoCellular.getSignalStrengthAsync) {
+    throw new UnavailabilityError('expo-cellular', 'getSignalStrengthAsync');
+  }
+  return await ExpoCellular.getSignalStrengthAsync();
+}
+
 /**
  * Checks user's permissions for accessing phone state.
  */
diff --git a/node_modules/expo-cellular/src/ExpoCellular.web.ts b/node_modules/expo-cellular/src/ExpoCellular.web.ts
index a7498ef..1f5dbab 100644
--- a/node_modules/expo-cellular/src/ExpoCellular.web.ts
+++ b/node_modules/expo-cellular/src/ExpoCellular.web.ts
@@ -51,4 +51,7 @@ export default {
   async getMobileNetworkCodeAsync(): Promise<string | null> {
     return null;
   },
+  async getSignalStrengthAsync(): Promise<number | null> {
+    return null;
+  },
 };
